woocommerce/woocommerce-ios
# [Product Multi-Selection] Add debug experimental toggle

Closes: #9095 


## Description
This PR adds a toggle to switch on/off Product Multi-Selection for developer builds (localDeveloper, and alpha), within the Experimental Features screen. This toggle will not show in production.

## Changes
- Added new `isProductMultiSelectionSwitchEnabled` flag to `GeneralAppSettings`
- Add a check to `BetaFeature` to make `isProductMultiSelectionSwitchEnabled` only available if the feature flag is on
- Inject the value of `Servicelocator.generalAppSettings.betaFeatureEnabled` into our view models,  rather than the feature flag.
- Updated `generateStorage/Storage/Model/Copiable/Models+Copiable.generated.swift` via rake generate
- Switch feature flag to `buildConfig == .localDeveloper || buildConfig == .alpha`

## Testing instructions
Scenario 1: Debug
- Confirm the app scheme is set to run in Build Configuration: `Debug` or `Release-Alpha`
- Run the app
- Go to Orders > `+` > `+ Add Product` > See that products can be tapped on a single-selection basis
- Go to Menu > Experimental features > Toggle on `Product Multi-Selection`
- Re-run the app (cmd + R, no need for re-build) in Xcode
- Go to Orders > `+` > `+ Add Products` > See that products can be tapped on a multi-selection basis

Scenario 2: Release
- Switch the app scheme to run `Build Configuration: Release`
- Re-build the app
- Go to Menu > Experimental features > Confirm that `Product Multi-Selection` does not appear as an option.

## Screenshots

<img width= 450 src="https://user-images.githubusercontent.com/3812076/224233779-4a1522d4-4610-42f8-98c6-5cecf2aabee3.png">
